File: app/schemas/schedule.py
from pydantic import BaseModel


class LessonResponse(BaseModel):
    time: str
    name: str
    teacher: str
    canceled: bool
    absent_count: int
    is_current: bool


class ScheduleResponse(BaseModel):
    date: str
    lessons: list[LessonResponse]


class OverrideUpdateRequest(BaseModel):
    date: str
    time: str
    new_name: str | None = None
    new_teacher: str | None = None
    is_canceled: int

File: frontend/src/types/schedule.ts
export interface Lesson {
  time: string;
  name: string;
  teacher: string;
  canceled: boolean;
  absent_count: number;
  is_current?: boolean;
}

export interface ScheduleResponse {
  date: string;
  lessons: Lesson[];
}

export interface OverrideUpdateRequest {
  date: string;
  time: string;
  new_name?: string | null;
  new_teacher?: string | null;
  is_canceled: number;
}
